packages/plugin-infra/src/types.ts

```diff
@@ -141,3 +146,16 @@ export interface I18NStringField {
     i18nKey?: string
     fallback: string
 }
+export interface PluginHost {
+    eth: EthStatusReporter
+    enabled: EnabledStatusReporter
+    signal?: AbortSignal
+}
+export interface EthStatusReporter {
+    current(): ChainId
```

Comment:
In the future, we are going to support other chains. Does it easy to disable this feature?

Reply:
This is a "tool" to statically declare what chain it supports and turn the plugin on or off. I think it's easy to extends/rewrite since it's declarative on the use site.